File: shared_schemas/food.py
from pydantic import BaseModel, ConfigDict

class FoodOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    food_code: str             # 식품코드
    food_name: str             # 식품명
    data_type: str             # 데이터구분명
    energy_kcal: float | None  # 에너지(kcal)
    water_g: float | None      # 수분(g)
    protein_g: float | None    # 단백질(g)
    fat_g: float | None        # 지방(g)
    carbohydrate_g: float | None  # 탄수화물(g)
    sugar_g: float | None      # 당류(g)
    fiber_g: float | None      # 식이섬유(g)
    calcium_mg: float | None   # 칼슘(mg)
    sodium_mg: float | None    # 나트륨(mg)
    vitamin_c_mg: float | None # 비타민C(mg)
    source_name: str | None    # 출처명

class FoodListParams(BaseModel):
    page: int = 1
    size: int = 20
    data_type: str | None = None  # 데이터구분명 필터

class FoodListResponse(BaseModel):
    items: list[FoodOut]
    total: int
    page: int
    size: int

File: spring_app/src/main/java/com/benchmark/entity/Food.java
package com.benchmark.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "foods")
public class Food {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "food_code", length = 20, unique = true, nullable = false)
    private String foodCode;

    @Column(name = "food_name", length = 200, nullable = false)
    private String foodName;

    @Column(name = "data_type", length = 50, nullable = false)
    private String dataType;

    @Column(name = "energy_kcal")
    private Double energyKcal;

    @Column(name = "water_g")
    private Double waterG;

    @Column(name = "protein_g")
    private Double proteinG;

    @Column(name = "fat_g")
    private Double fatG;

    @Column(name = "carbohydrate_g")
    private Double carbohydrateG;

    @Column(name = "sugar_g")
    private Double sugarG;

    @Column(name = "fiber_g")
    private Double fiberG;

    @Column(name = "calcium_mg")
    private Double calciumMg;

    @Column(name = "sodium_mg")
    private Double sodiumMg;

    @Column(name = "vitamin_c_mg")
    private Double vitaminCMg;

    @Column(name = "source_name", length = 200)
    private String sourceName;

    // Getters
    public Long getId() { return id; }
    public String getFoodCode() { return foodCode; }
    public String getFoodName() { return foodName; }
    public String getDataType() { return dataType; }
    public Double getEnergyKcal() { return energyKcal; }
    public Double getWaterG() { return waterG; }
    public Double getProteinG() { return proteinG; }
    public Double getFatG() { return fatG; }
    public Double getCarbohydrateG() { return carbohydrateG; }
    public Double getSugarG() { return sugarG; }
    public Double getFiberG() { return fiberG; }
    public Double getCalciumMg() { return calciumMg; }
    public Double getSodiumMg() { return sodiumMg; }
    public Double getVitaminCMg() { return vitaminCMg; }
    public String getSourceName() { return sourceName; }
}
